src/test/scala/vsys/api/http/ApiErrorSpecification.scala

```diff
@@ -0,0 +1,246 @@
+
+class ApiErrorSpecification extends PropSpec with PropertyChecks with GeneratorDrivenPropertyChecks with Matchers with TransactionGen {
+  property("ApiError should be thrown if alias not exist") {
+    val alias = Alias.fromString("alias:T:sawesha").right.get
+    val err = ValidationError.AliasNotExists(alias)
+    val result = ApiError.fromValidationError(err)
+    result.id should be(AliasNotExists(alias).id)
+    result.code should be(StatusCodes.NotFound)
+  }
+  property("UnsupportedTransactionType should result in CustomValidationError with msg 'UnsupportedTransactionType'") {
+    val err = ValidationError.UnsupportedTransactionType
+    val result = ApiError.fromValidationError(err)
+    result.id should be(CustomValidationError(result.message).id)
+    result.code should be(StatusCodes.BadRequest)
+    result.message should be("UnsupportedTransactionType")
+  }
+
+  property("AccountBalanceError should get err which value is err message list"){
+    forAll(accountGen, accountGen) { (account1, account2) => 
+      var errs: Map[Address, String] = Map()
+      val addr1 = account1.toAddress
+      val addr2 = account2.toAddress
+      errs += (addr1 -> "No enough money for output")
+      errs += (addr2 -> "Account is locked")
+      val err = ValidationError.AccountBalanceError(errs)
+      val result = ApiError.fromValidationError(err)
+      result.id should be(CustomValidationError(result.message).id)
+      result.code should be(StatusCodes.BadRequest)
+      result.message should be("No enough money for output, Account is locked")
+    }
+  }
+
+
+  property("OrderValidationError should cause CustomValidationError which specified msg") {
+    forAll(orderGen) { order =>
+      val testOrder = Order.parseBytes(order.bytes).get
+      val err = ValidationError.OrderValidationError(testOrder, "order validation throw error")
+      val result = ApiError.fromValidationError(err)
+      result.id should be(CustomValidationError(result.message).id)
+      result.code should be(StatusCodes.BadRequest)
+      result.message should be("order validation throw error")
+    }
+  }
+
+  property("Mistiming should cause ApiError.Mistiming eror") {
+    val err = ValidationError.Mistiming("Mistiming error")
+    val result = ApiError.fromValidationError(err)
+    result.id should be(Mistiming.Id)
+    result.code should be(StatusCodes.BadRequest)
+    result.message should be("Mistiming error")
+  }
+
+  property("DbDataTypeError should cause error which id is 504 and specified message ") {
+    val dbDataType = "SuperNode"
+    val err = ValidationError.DbDataTypeError(dbDataType) 
+    val result = ApiError.fromValidationError(err)
+    result.id should be(InvalidDbDataType(dbDataType).id)
+    result.code should be(StatusCodes.BadRequest)
+    result.message should be("invalid database datatype SuperNode")
+  }
+
+  property("WrongFeeScale should cause err which id is 114") {
+    forAll(feeScaleGen) { feeScale => 
```

Comment:
in order to cover the all situations, here the input can be more general, not only the correct feeScale range. A `Short` type input can be used here directly. similarly in later test.